invoicing/UI/desktop/home.js

```diff
@@ -4,17 +4,23 @@ ypnUi.define('Home', {
 
 ypn.onready((_callback) => {
     YContext.on("packageready", function (_err) {
-        // debugger;
-        if (_err)
+        if (_err) {
             console.error(_err.message);
-        else
-            session.dbserver.connect("nsC", 'INVOICING', function (_err, _cnx) {
-                if (_err) console.error('failled to get the connection');
-                else {
-                    YContext.emit("connectionsready");
-                    session.defaultConnection = 'nsC';
-                }
-            });
+            return;
+        }
+
+        YContext.on("connectionsready", () => {
+            //fixme: emit contextready to trigger currency loader
+            //YContext.emit('contextready');
+        });
+
+        session.dbserver.connect("nsC", "INVOICING").then( () => {
```

Comment:
Should we update to async/await ?

Reply:
yep, eventually. Not in this PR. Don't want to change too much.